packages/admin-portal/src/components/iaas/index.tsx

```diff
@@ -0,0 +1,198 @@
+              getResults={(company: string) =>
+                fetchDevelopersList({ company }).then((developers) => developers?.data ?? [])
+              }
+              getResultLabel={(result) => `${result.company} -  ${result.name}`}
+              getResultValue={(result) => result.id ?? ''}
+              placeholder="Search developer organisations"
+            />
+          </InputWrap>
+        </FormLayout>
+      </form>
+      {pipelinesLoading ? (
+        <Loader />
+      ) : (
+        <>
+          <Table
+            className={elMb11}
+            numberColumns={4}
+            rows={pipelines?.items.map(({ name, id, buildStatus, repository }) => ({
+              cells: [
+                {
+                  label: 'Name',
+                  value: name,
+                  icon: 'apiInfographic',
+                  cellHasDarkText: true,
+                  narrowTable: {
+                    showLabel: true,
+                  },
+                },
+                {
+                  label: 'Build Status',
+                  value: (
+                    <>
+                      <StatusIndicator intent={buildStatusToIntent(buildStatus as string)} />
+                      {buildStatusToReadable(buildStatus as string)}
+                    </>
+                  ),
+                  narrowTable: {
+                    showLabel: true,
+                  },
+                },
+                {
+                  label: 'Repository',
+                  value: repository,
+                  narrowTable: {
+                    showLabel: true,
+                  },
+                },
+              ],
+              ctaContent: {
+                headerContent: 'Delete Pipeline',
+                icon: pipelineNotDeletable.includes(buildStatus as string) ? undefined : 'trashSystem',
+                onClick: pipelineNotDeletable.includes(buildStatus as string)
+                  ? undefined
+                  : handleOpenModal(setAppId, openModal, id),
+              },
+            }))}
+          />
+          <Modal title="Delete Pipeline">
+            <BodyText hasGreyText>
+              Are you sure you want to delete this pipeline? This will tear down any infrastructure you have provisioned
+              and cannot be recovered.
+            </BodyText>
+            <ButtonGroup alignment="center">
+              <Button fixedWidth intent="low" onClick={closeModal}>
+                Close
+              </Button>
+              <Button
+                fixedWidth
+                intent="danger"
+                onClick={handleDeletePipeline(deletePipeline, closeModal, refreshPipelines)}
+              >
+                Delete
+              </Button>
+            </ButtonGroup>
+          </Modal>
+          {pipelines && (
+            <Pagination
+              currentPage={pipelines.meta.currentPage ?? 1}
+              numberPages={pipelines.meta.totalPages ?? 1}
+              callback={setPage}
```

Comment:
think you need a useEffect on setPage to reload data?

Reply:
No, it works correctly